tests/test_update.py

```diff
@@ -663,12 +663,28 @@ def test_signed_updates(
                 shutil.rmtree(tmpdir, ignore_errors=True)
 
         put_no_sftp("image.mender", connection, remote="/data/image.mender")
+
+        # mender-convert'ed images don't have transient mender.conf
+        device_has_mender_conf = (
+            connection.run("test -f /etc/mender/mender.conf", warn=True).return_code
+            == 0
+        )
+        # mender-convert'ed images don't have this directory, but the test uses
+        # it to save certificates
+        connection.run("mkdir -p /data/etc/mender")
+
         try:
-            # Update key configuration on device.
-            connection.run(
-                "cp /etc/mender/mender.conf /data/etc/mender/mender.conf.bak"
-            )
-            get_no_sftp("/etc/mender/mender.conf", connection)
+            # Get configuration from device or create an empty one
+            if device_has_mender_conf:
+                connection.run(
+                    "cp /etc/mender/mender.conf /data/etc/mender/mender.conf.bak"
+                )
+                get_no_sftp("/etc/mender/mender.conf", connection)
```

Comment:
shall we check here if the transfer was successful, `get_no_sftp` uses `conn.local` but there is no try/except there.

Reply:
This part has not changed.

Besides, we do check before if the device has `mender.conf`. If so, this `get_no_sftp` call shall not fail.

Reply:
ok, but put_ does have a try block, and get_ does not, perhaps it should then.

Reply:
I see now. Well, `put_` has a try/catch only to print a nicer error and then re-raise the exception. That to me is quite useless so, if any, I would remove that.